### PR TITLE
Conditionally show jump to original location option (#1749)

### PR DESCRIPTION
**Pull request by jbhoosreddy**
Associated Issue: #1749

### Summary of Changes

* Add a handler to show jump to original location only if the source has an original source.

### Test Plan

Visual testing

Example test plan:

- [x] Right click in Editor on a source mapped source.
- [x] Should give option for "Jump to Original Source".
- [x] Right click in Editor on a source mapped original source.
- [x] Should give option for "Jump to Generated Source".
- [x] Right click in Editor on a non source mapped source.
- [x] There should not be an option for "Jump to Original Source".

#### When it shouldn't appear
screen shot 2017-01-19 at 9 11 02 pm

#### When it should appear
screen shot 2017-01-19 at 9 11 30 pm

screen shot 2017-01-19 at 9 11 43 pm

### CODE REVIEW

File: src/utils/source-map.js [hunk 1]
```diff
@@ -34,6 +36,10 @@ function shouldSourceMap(): boolean {
   return prefs.clientSourceMapsEnabled;
 }
 
+function hasMappedSource(source: Source): boolean {
+  return !isOriginalId(source.id);
```
Comment (jasonLaster) [inline src/utils/source-map.js]: hmm, two things:

1. I think this should be `source.get("id")` -> i saw an error in the console
2. I think we first need to check if it's original or generated and then check for the other id... right now we assume its generated

Comment (jbhoosreddy) [inline src/utils/source-map.js]: oh sorry for the messup. it's still wip. I'll ping you again once I'm ready with this.

**Comment by jbhoosreddy** [comment]
@jasonLaster up for review now.

File: src/components/Editor.js [hunk 2]
```diff
@@ -136,8 +136,8 @@ const Editor = React.createClass({
 
     const sourceLocation = {
       sourceId: this.props.selectedLocation.sourceId,
-      line: line,
-      column: ch
+      line: line + 1,
+      column: ch + 1
     };
```
Comment (jbhoosreddy) [inline src/components/Editor.js]: the `source-map` library expects numeric values greater than 0.

**Comment by jasonLaster** [comment]
Thanks @jbhoosreddy! This is pretty cool. I love when a feature is UI + redux.

I'm not sure if i'm a fan of using state in this way to handle the async side of the code, so i tried making the handler async.

* the async state seemed to scatter some of the logic so it was tough to follow
* i think this is the first async event handler, which is kinda cool

**Comment by jbhoosreddy** [comment]
This is also one of the things I was considering but I wasn't sure if we wanted to go that way. I think I was thinking maybe it'd slow down the context menu appearance. Like perceived lag for the user as it is fetching this information on the event. @jasonLaster. The fact that it'll be mapped should be constant for each source.

That being said, I didn't see any lag when I tried it with an async function. What do you think?

**Comment by jasonLaster** [comment]
I don't see any lag either. I think if there were some then I'd agree with you, but it's pretty snappy.

Communicating with a worker is now [~1ms](https://hacks.mozilla.org/2015/07/how-fast-are-web-workers/) and doing the lookup is fast.

**Comment by jasonLaster** [comment]
@jbhoosreddy I async'd `hasMappedSource` as well. Let me know what you think.

**Comment by jasonLaster** [comment]
merging when it goes green

**Pull request merged**